Model/Data/OrderUpdate.php

```diff
@@ -249,9 +249,9 @@ private function dataValidation(Order $order, DataObject $amwalOrderData)
             $this->sendAdminEmail($order, 'Order (%s) needs Attention', $this->dataValidationMessage($order->getIncrementId(), 'order_currency_code', 'default_currency_code', $order->getOrderCurrencyCode(), self::DEFAULT_CURRENCY_CODE));
             throw new \Exception(sprintf('Order (%s) %s does not match Amwal Order %s (%s != %s)', $order->getIncrementId(), 'order_currency_code', 'default_currency_code', $order->getOrderCurrencyCode(), self::DEFAULT_CURRENCY_CODE));
         }
-        if (floatval($order->getBaseGrandTotal()) != floatval($amwalOrderData->getTotalAmount())) {
-            $this->sendAdminEmail($order, 'Order (%s) needs Attention', $this->dataValidationMessage($order->getIncrementId(), 'base_grand_total', 'total_amount', $order->getBaseGrandTotal(), $amwalOrderData->getTotalAmount()));
-            throw new \Exception(sprintf('Order (%s) %s does not match Amwal Order %s (%s != %s)', $order->getIncrementId(), 'base_grand_total', 'total_amount', $order->getBaseGrandTotal(), $amwalOrderData->getTotalAmount()));
+        if (floatval($order->getGrandTotal()) != floatval($amwalOrderData->getTotalAmount())) {
```

Comment:
Let's use this way for casting to float (~6x faster)
```suggestion
        if ((float)$order->getGrandTotal() != (float)$amwalOrderData->getTotalAmount()) {
```